presto-hive/src/main/java/com/facebook/presto/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -183,13 +190,17 @@ private Estimate calculateNullsCountForPartitioningKey(HiveColumnHandle partitio
             return Estimate.unknownValue();
         }
 
-        return new Estimate(partitions.stream()
+        double estimatedTotalRowsCount = rowsPerPartition.getAsDouble() * partitions.size();
+        if (estimatedTotalRowsCount == 0.0) {
+            return Estimate.zeroValue();
+        }
+        double estimatedNullsCount = partitions.stream()
```

Comment:
This is counted as long so perhaps make `estimatedNullsCount` a `long`, deferring cast to more visible place.